zubhub_frontend/zubhub/src/components/hamburger_menu/HamburgerSidebar.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import styles from '../../assets/js/styles/components/hamburger_menu/hamburgerSidebarStyles.js';
+import commonStyles from '../../assets/js/styles';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  logout,
+  handleHamburgerMenuClose,
+} from '../../views/pageWrapperScripts';
+import clsx from 'clsx';
+import cn from 'classnames';
+
+import {
+  Typography,
+  Drawer,
+  Link,
+  Avatar,
+  MenuItem,
+} from '@material-ui/core';
+
+const useStyles = makeStyles(styles);
+const useCommonStyles = makeStyles(commonStyles);
+
+const HamburgerSidebar = ({
+  hamburgerMenuOpen,
+  setState,
+  t,
+  props,
+}) => {
+  const classes = useStyles();
+  const common_classes = useCommonStyles();
+  const handleSetState = obj => {
+    if (obj) {
+      Promise.resolve(obj).then(obj => {
+        setState(state => ({ ...state, ...obj }));
+      });
+    }
+  };
```

Comment:
I don't think this is necessary. You can use a normal state variable with the ```useState``` hook and use that to control when the sidebar shows up or not. One thing you could also try is moving the ```HamburgerSidebar``` component into the ```HamburgerButton``` component and just calling that from ```PageWrapper``` so it's a little bit cleaner. This would also allow you to isolate the state variable into that component without having to use it in ````PageWrapper```.